File: lib/utils/token.py
class Token:
    def __init__(self, kind, lexeme, location):
        self.kind = kind
        self.lexeme = lexeme
        self.location = location

    def __repr__(self):
        return f"Token({self.kind}, '{self.lexeme}', Location({self.location[0]}, {self.location[1]}, {self.location[2]}))"

File: src/utils/token.rs
use crate::utils::token_kind::TokenKind;

#[derive(Debug)]
pub struct Token {
    pub kind: TokenKind,
    pub lexeme: String,
    pub location: [usize; 3],
}

impl Token {
    pub fn new(kind: TokenKind, lexeme: String, location: [usize; 3]) -> Self {
        Token {
            kind,
            lexeme,
            location,
        }
    }
}
